src/schrodinger/rdkit_extensions/helm/monomer_coordgen.cpp

```diff
@@ -108,17 +128,69 @@ lay_out_chain(RDKit::ROMol& polymer, const RDKit::Atom* start_monomer,
+                next_monomer_to_layout = neighbor;
             } else {
-                // branch monomer - placed with a y offset at the same x coord
-                double y_offset = branch_direction == UP ? MONOMER_BOND_LENGTH
-                                                         : -MONOMER_BOND_LENGTH;
-                RDGeom::Point3D pos(x_pos, start_pos.y + y_offset, start_pos.z);
-                conformer.setAtomPos(neighbor->getIdx(), pos);
-                neighbor->setProp(MONOMER_PLACED, true);
+                branches.push_back(neighbor);
             }
         }
-        x_pos += chain_dir == LTR ? MONOMER_BOND_LENGTH : -MONOMER_BOND_LENGTH;
+        // list the available directions for branch monomers, in order of
+        // preference
+        std::vector<BranchDirection> available_directions;
+        // first try the main branch direction
+        if ((!bonded_to_child_polymer &&
+             branch_direction == BranchDirection::DOWN) ||
+            (!bonded_to_parent_polymer &&
+             branch_direction == BranchDirection::UP)) {
+            available_directions.push_back(branch_direction);
+        }
+        // if this is the first or last direction, we can branch along
+        // the chain
+        if (!next_monomer_to_layout) {
+            available_directions.push_back(chain_dir == ChainDirection::LTR
+                                               ? BranchDirection::RIGHT
+                                               : BranchDirection::LEFT);
+        }
+        if (!last_laidout_monomer) {
+            available_directions.push_back(chain_dir == ChainDirection::LTR
+                                               ? BranchDirection::LEFT
+                                               : BranchDirection::RIGHT);
+        }
+        // as a last resort, try the opposite of the main branch direction
+        if (!bonded_to_parent_polymer &&
+            branch_direction == BranchDirection::DOWN) {
+            available_directions.push_back(BranchDirection::UP);
+        } else if (!bonded_to_child_polymer &&
+                   branch_direction == BranchDirection::UP) {
+            available_directions.push_back(BranchDirection::DOWN);
+        }
+        auto first_available_direction = available_directions.begin();
+
+        for (auto branch_monomer : branches) {
+            RDGeom::Point3D pos(x_pos, start_pos.y, start_pos.z);
+            switch (*first_available_direction++) {
+                case BranchDirection::UP:
+                    pos.y += MONOMER_BOND_LENGTH;
+                    break;
+                case BranchDirection::DOWN:
+                    pos.y -= MONOMER_BOND_LENGTH;
+                    break;
+                case BranchDirection::LEFT:
+                    pos.x -= MONOMER_BOND_LENGTH;
+                    break;
+                case BranchDirection::RIGHT:
+                    pos.x += MONOMER_BOND_LENGTH;
+                    break;
+            }
```

Comment:
You could also pull this out into something like `static void move_point_by_distance(RDGeom::Point3D& pos, const BranchDirection dir, const qreal& distance)` to make this logic a little more readable.